### PR TITLE
feat: config axios

### PR DESCRIPTION
**Pull request by TranQuangPhi**

### CODE REVIEW

File: template/src/utilities/networking.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      }
+
+      default: {
+        return (axios.defaults.headers.common["Authorization"] = token);
+      }
+    }
+  };
+
+export function getRequest(api, config) {
+  return instance.get(api, config).then(response => {
```
Comment (hanzoow) [inline template/src/utilities/networking.js:35]: Need to add debug line, e.g __DEV__ && console.log("request>>>>", api, config)

Comment (hanzoow) [inline template/src/utilities/networking.js:35]: Need to add debug line, e.g __DEV__ && console.log("response>>>>", api, response)

Comment (hanzoow) [inline template/src/utilities/networking.js:35]: For all request methods like post, put, delete, ...

File: template/src/utilities/networking.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+export function postRequest(api, body, config) {
+  if (typeof body === 'object' && body.constructor !== FormData) body = JSON.stringify(body)
+
+  return instance.post(api, body, config).then(response => {
+    return {data: response.data, status: response.status}
+  }).catch(error => {
+    return {data: null, status: error.response.status}
+  })
+}
+
```
Comment (hoanganhle1996) [inline template/src/utilities/networking.js:51]: Willing to have postFormData function

File: template/src/utilities/axiosInstance.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import axios from 'axios'
+
+const instance = axios.create({
+  baseURL: 'MY_URL',
+  timeout: 1000,
+  withCredentials: false,
```
Comment (hoanganhle1996) [inline template/src/utilities/axiosInstance.js:6]: timeout to small value, should be 60000

File: template/src/utilities/axiosInstance.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import axios from 'axios'
+
+const instance = axios.create({
+  baseURL: 'MY_URL',
+  timeout: 1000,
```
Comment (hoanganhle1996) [inline template/src/utilities/axiosInstance.js:5]: timeout to small value, should be 60000

File: template/src/utilities/networking.js (hunk 1 above)
Comment (TranQuangPhi) [reply template/src/utilities/networking.js:35]: Hi @hanzoow, I'll create a ticket to integrate the log, but not for this PR

File: template/src/utilities/axiosInstance.js (hunk 4 above)
Comment (TranQuangPhi) [reply template/src/utilities/axiosInstance.js]: done

File: template/src/utilities/networking.js (hunk 2 above)
Comment (TranQuangPhi) [reply template/src/utilities/networking.js:51]: Hi @hoanganhle1996, This function was added. Please let me know if there are any problems

File: template/src/utilities/axiosInstance.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import axios from 'axios'
+
+const instance = axios.create({
+  baseURL: 'MY_URL',
```
Comment (honghoangsts) [inline template/src/utilities/axiosInstance.js:4]: Could we get it in env config

File: template/src/utilities/networking.js [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  instance.defaults.headers.common[property] = data;
+};
+
+export function setBaseURL (baseURL) {
+  instance.defaults.baseURL = baseURL;
+};
+
+export function setToken (token, type = null) {
+    switch (type) {
+      case "Paymentez": {
```
Comment (honghoangsts) [inline template/src/utilities/networking.js:13]: Could we create `enum` for this value. Don't hard code the string

File: template/src/utilities/networking.js [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    }
+    config = {
+      ...config,
+      headers,
+    }
+    return instance.post(api, data, config).then(response => {
+      return {data: response.data, status: response.status}
+    }).catch(error => {
+      return {data: null, status: error.response.status}
+    })
```
Comment (honghoangsts) [inline template/src/utilities/networking.js:68]: Can we reuse code from postRequest?

File: template/src/utilities/networking.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+  return instance.patch(api, body, config).then(response => {
+    return {data: response.data, status: response.status}
+  }).catch(error => {
+    return {data: null, status: error.response.status}
+  })
+}
+
+export function deleteRequest(api, config) {
+  if (typeof body === 'object' && body.constructor !== FormData) body = JSON.stringify(body)
```
Comment (honghoangsts) [inline template/src/utilities/networking.js:95]: Could we make it in common so it not duplicate the code

**Review by honghoangsts — CHANGES_REQUESTED** [review]

**Comment by honghoangsts** [comment]
I think we should put it to `/services`

File: template/src/utilities/axiosInstance.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+  // Do something before request is sent
+  return config
+}, error => Promise.reject(error))
+
+instance.interceptors.response.use(response => {
+  // Do something with response data
+  return response
+}, error => Promise.reject(error))
+
+export default instance
```
Comment (ThinhKimVo) [inline template/src/utilities/axiosInstance.js:25]: follow ESlint rule **new line of the end of file**

File: template/src/utilities/networking.js [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+  return instance.delete(api, config).then(response => {
+    return {data: response.data, status: response.status}
+  }).catch(error => {
+    return {data: null, status: error.response.status}
+  })
+}
```
Comment (ThinhKimVo) [inline template/src/utilities/networking.js:102]: follow ESlint rule **new line of the end of file**

File: template/src/utilities/axiosInstance.js (hunk 5 above)
Comment (ThinhKimVo) [reply template/src/utilities/axiosInstance.js:4]: should create a **service** folder and put it inside

File: template/src/utilities/networking.js [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+export function setHeader (property, data) {
+  instance.defaults.headers.common[property] = data;
+};
+
+export function setBaseURL (baseURL) {
+  instance.defaults.baseURL = baseURL;
+};
+
+export function setToken (token, type = null) {
```
Comment (ThinhKimVo) [inline template/src/utilities/networking.js:11]: setAuthHeader

File: template/src/utilities/networking.js [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+    return {data: response.data, status: response.status}
+  }).catch(error => {
+    return {data: null, status: error.response.status}
+  })
+}
+
+export function patchRequest(api, body, config) {
+  if (typeof body === 'object' && body.constructor !== FormData) body = JSON.stringify(body)
+
+  return instance.patch(api, body, config).then(response => {
```
Comment (ThinhKimVo) [inline template/src/utilities/networking.js:87]: implement common func (api, body, config) => {data: response.data, status: response.status} to reuse for all API request, included refresh token logic

File: template/src/utilities/axiosInstance.js (hunk 5 above)
Comment (TranQuangPhi) [reply template/src/utilities/axiosInstance.js:4]: done

File: template/src/utilities/networking.js (hunk 6 above)
Comment (TranQuangPhi) [reply template/src/utilities/networking.js:13]: done

File: template/src/utilities/networking.js (hunk 7 above)
Comment (TranQuangPhi) [reply template/src/utilities/networking.js:68]: done

File: template/src/utilities/networking.js (hunk 8 above)
Comment (TranQuangPhi) [reply template/src/utilities/networking.js:95]: done

File: template/src/utilities/axiosInstance.js (hunk 9 above)
Comment (TranQuangPhi) [reply template/src/utilities/axiosInstance.js:25]: done

File: template/src/utilities/networking.js (hunk 10 above)
Comment (TranQuangPhi) [reply template/src/utilities/networking.js:102]: done

File: template/src/utilities/networking.js (hunk 11 above)
Comment (TranQuangPhi) [reply template/src/utilities/networking.js:11]: I think setToken name is the most obvious name. This name shouldn't be changed.

**Pull request closed without merging**